src/Microsoft.Health.Fhir.Shared.Core/Features/Search/BundleFactory.cs

```diff
@@ -83,6 +83,7 @@ private ResourceElement CreateBundle(SearchResult result, Bundle.BundleType type
             {
                 var operationOutcome = new OperationOutcome
                 {
+                    Id = _fhirRequestContextAccessor.FhirRequestContext.CorrelationId,
```

Comment:
I don't think this makes sense when the bundle will have the same id set below (https://github.com/microsoft/fhir-server/pull/1385/files#diff-038e862a9e90e7e3da159fb2485bc1017c36fbc313bca2f4bd8d2c3052ad2f88R116)

Reply:
[https://www.hl7.org/fhir/resource-definitions.html#Resource.id](https://www.hl7.org/fhir/resource-definitions.html#Resource.id)

```
Comments 
The only time that a resource does not have an id is when it is being submitted to the server using a create operation.
```

My interpretation of standard - we need to return id for every resource. I know it's same as id in bundle, (I'm actually not sure is it ok for two resources have same id or not, couldn't find clarification in standard), but I think it's better than return nothing.

---
In reply to: [525361013](https://github.com/microsoft/fhir-server/pull/1385#discussion_r525361013) [](ancestors = 525361013)

Reply:
I think that part is ok, for example `Patient/example` and `Observation/example` are both valid with the same "id". I guess under that pretense it makes sense to leave it in